File: sdk/go-guest/examples/random.go
package export_tests_random_api

import random "wit_component/lsf/random"

func Run(which uint32, _ string, _ uint64) uint64 {
	switch which {
	case 0:
		return uint64(len(random.Bytes(32).Ok()))
	case 1:
		random.U64Value().Ok()
		return 8
	case 2:
		if random.Bytes(^uint32(0)).Err().Tag() != random.RandomErrorInvalidLength {
			panic("random length changed")
		}
		return 10
	default:
		panic("unknown random probe")
	}
}

File: sdk/typescript-guest/examples/random.ts
import type * as Contract from '../generated/interfaces/tests-random-api.js';
import * as random from '../vendor/lsf/sdk/typescript-guest/capabilities/random.js';
import { unwrap } from '../vendor/lsf/sdk/typescript-guest/capabilities/result.js';
export const api: typeof Contract = {
  run(which) {
    if (which === 1) { unwrap(random.u64()); return 8n; }
    const result = random.bytes(which === 2 ? 0xffff_ffff : 32);
    if (result.tag === 'err' && result.val.tag === 'invalid-length') return 10n;
    return BigInt(unwrap(result).length);
  },
};
